crates/p2p/src/swarm/behavior.rs

```diff
@@ -67,6 +68,20 @@ impl Behaviour {
                     .validation_mode(gossipsub::ValidationMode::Permissive)
                     .validate_messages()
                     .max_transmit_size(MAX_TRANSMIT_SIZE)
+                    // Avoids spamming the network and nodes with messages
+                    .idontwant_on_publish(true)
+                    // We want a unique message id for each message, so we use the hash of the
+                    // message data instead of the default one, that is the concatenation of the
+                    // PeerId and the sequence number of the message.
+                    //
+                    // NOTE(@storopoli): I don't trust the default one, since we are not using the
+                    //                   LibP2P's Message template, hence the sequence number might
+                    //                   not exist, and in that case it is always be set to 0 by
+                    //                   default.
+                    .message_id_fn(|msg| {
+                        let hash = hash(msg.data.as_ref());
```

Comment:
Recommend using a domain separator here as good practice.

Reply:
Check 0429cb59f45c25554bb6654d64fe623eda5800f1

Reply:
Given discussion elsewhere, domain separation is not needed here. Suggestion withdrawn!